app/controllers/learning_hours_controller.rb

```diff
@@ -3,7 +3,15 @@ class LearningHoursController < ApplicationController
   after_action :verify_authorized, except: :index # TODO add this back and fix all tests
 
   def index
-    @learning_hours = LearningHour.where(user_id: current_user.id)
+    authorize LearningHour
+
+    @learning_hours = if current_user.casa_admin?
+      LearningHour.all_volunteers_learning_hours
+    elsif current_user.supervisor?
+      LearningHour.supervisor_volunteers_learning_hours(current_user.id)
+    else # for volunteer user
+      LearningHour.where(user_id: current_user.id)
+    end
```

Comment:
Would you mind putting this in the `LearningHour` policy `Scope`? You can see `app/policy/user_scope.rb` for an example of defining the Scope. Then you can call `@learning_hours = policy_scope(LearningHour)` here in the `index` method. Pundit will automatically pass in `current_user` when calling the scope.

[Documentation reference](https://github.com/varvet/pundit#scopes)

Reply:
Added scope for LearningHour and related spec to test the scenario.